### ISSUE TITLE
Search results missing some packages

### ISSUE THREAD

**Issue by rmanne** [NixOS/nixos-search#708]
### Describe the bug
I'm not sure how many packages are impacted, but I can confirm that at least 1 package is missing from search. Direct link to the [search results](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=yabai).

Couldn't find a bug template, so copied over the one from nixpkgs and removed some irrelevant sections.

### Steps To Reproduce
1. Search for "yabai" ([direct link](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=yabai))
2. Observe that there are no packages found
3. Run `nix-shell -p yabai`
4. Observe it works, that there exists a package with the name "yabai". (Run `which yabai` to confirm such a binary has been installed.)

### Expected behavior
Searching for "yabai" should display the corresponding package, found here: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/os-specific/darwin/yabai/default.nix#L8

search does seem to find a related package which is configured in a similar way, in around the same path: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/os-specific/darwin/skhd/default.nix#L43

I'm not sure why the latter shows up in search.nixos.org but the former doesn't.

### Screenshots
<img width="1211" alt="Screenshot 2023-11-29 at 21 12 38" src="https://github.com/NixOS/nixos-search/assets/1724749/811be46f-2d73-450c-83b0-90a5ca64b2bc">

### Metadata

```console
> nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(rmanne): `"darwin, nixpkgs-23.11pre519165.cddebdb60de3"`
 - channels(root): `"darwin, nixpkgs-23.11pre519165.cddebdb60de3"`
 - nixpkgs: `/Users/rmanne/.nix-defexpr/channels/nixpkgs`
```

**Comment by iFreilicht** [comment]
Indeed, all packages that are not available on Linux are not shown.

**Comment by ChocolateLoverRaj** [comment]
Another package that isn't shown: `atk`.

> Indeed, all packages that are not available on Linux are not shown.

But this one is actually available on Linux

**Comment by iFreilicht** [comment]
> Another package that isn't shown: `atk`.

Where is that even defined? I can only find atkmm in the nixpkgs source. There's no directory named "atk" (`find . -type d -name atk` returns nothing), and no `.nix` file containing the string `name = "atk"`. How can you even install that package?

**Comment by ChocolateLoverRaj** [comment]
> How can you even install that package

I can specify the package in `buildInputs` in a `mkDerivation` (I use NixOS)

**Comment by ghpzin** [comment]
`atk` is an "alias" (but not in aliases.nix atm):
https://github.com/NixOS/nixpkgs/blob/5f0b88ca430f18701424c50a4ccbff8fc7e186ba/pkgs/top-level/all-packages.nix#L21202-L21203
But package that it is an alias of `at-spi2-core` is not shown in search either, only another of its' aliases `at-spi2-atk`:
https://search.nixos.org/packages?channel=unstable&show=at-spi2-atk&from=0&size=50&sort=relevance&type=packages&query=at-spi2-core
https://github.com/NixOS/nixpkgs/blob/5f0b88ca430f18701424c50a4ccbff8fc7e186ba/pkgs/top-level/all-packages.nix#L20047-L20048
